wal.go

```diff
@@ -0,0 +1,105 @@
+package azureauth
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/mitchellh/mapstructure"
+)
+
+const (
+	walAppKey          = "appCreate"
+	walRotateRootCreds = "rotateRootCreds"
+)
+
+// Eventually expire the WAL if for some reason the rollback operation consistently fails
+var maxWALAge = 24 * time.Hour
+
+func (b *azureAuthBackend) walRollback(ctx context.Context, req *logical.Request, kind string, data interface{}) error {
+	switch kind {
+	case walAppKey:
+		return b.rollbackAppWAL(ctx, req, data)
```

Comment:
We shouldn't be creating or rolling back applications in the auth method. Suggest to remove everything related to rolling back applications from this.

Reply:
Ah nice, that was a bit that was confusing me; thanks for clarifying! Removed all delete application logic from this flow (also simplified our interface 😅)